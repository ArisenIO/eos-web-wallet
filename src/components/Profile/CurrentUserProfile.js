//@flow
import Profile from "./Profile";
import withProfile from "../../containers/Profile/index";
import withUserId from "../../containers/UserInfo/index";

export default withUserId(withProfile(Profile));
