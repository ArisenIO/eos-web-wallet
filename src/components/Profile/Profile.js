//@flow
import * as React from "react";
import cx from "classnames";
import css from "./styles.module.scss";

type Props = {
  imageUrl?: string,
  currentLocation?: string,
  displayName?: string,
  status?: string,
  websiteUrl?: string
};

const Profile = ({
  imageUrl = "/images/user.png",
  currentLocation = "Unknown",
  displayName = "Display Name",
  status = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
  websiteUrl = "http://twitter.com/"
}: Props) => (
  <div className={cx("box", css.profile)}>
    <div className="columns is-variable is-2 is-vcentered">
      <div className="column is-narrow">
        <img src={imageUrl} className={css.thumbnail} alt={displayName} />
      </div>
      <div className="column">
        <div>
          <div className={cx("heading is-6", css.heading)}>{displayName}</div>
          <div className={cx("title", css.title)}>{currentLocation}</div>
          <a href={websiteUrl} target="_blank">{websiteUrl}</a>
        </div>
      </div>
    </div>
    <div>
      <p>{status}</p>
    </div>
  </div>
);

export default Profile;
