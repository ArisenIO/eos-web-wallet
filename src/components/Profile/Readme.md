Profile example:

```js
<Background className="prefrences-form-example content">
  <Profile
    name="Beth"
    location="Creve Couer, MO"
    image={{ url: "" }}
    status="Blockchain Rocks"
  />
</Background>
```
