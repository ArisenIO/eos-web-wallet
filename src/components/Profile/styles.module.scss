@import "../../styles/custom_variables.scss";

.profile {
  .thumbnail {
    border: solid 8px $orange;
    width: 125px;
    height: 125px;
    border-radius: 75px;
  }
  .heading {
    font-size: 22px;
    font-weight: bold;
    text-transform: capitalize;
  }
  .title {
    font-size: 18px;
    color: $white;
    margin-bottom: .5rem;
  }
}