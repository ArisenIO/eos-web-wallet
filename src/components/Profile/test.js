import * as React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";
import Profile from "./";

describe("<Profile />", () => {
  it("renders without crashing", () => {
    const staticContext = {};
    const profile = {
      imageUrl: "src",
      currentLocation: "Friday Harbor, WA",
      displayName: "Rick Sanchez",
      status: "active"
    };

    const tree = renderer
      .create(
        <StaticRouter location="/" context={staticContext}>
          <Profile {...profile} />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
