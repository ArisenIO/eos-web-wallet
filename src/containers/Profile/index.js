import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import Profile from "components/Profile";
import {
  selectWalletUserImage,
  selectWalletUserName
} from "../../redux-modules/user/user-selectors";

// TODO genericize this to network call to specific user
const mapStateToProps = state => ({
  displayName: selectWalletUserName(state),
  imageUrl: selectWalletUserImage(state)
});

const Container = connect(mapStateToProps)(Profile);

export default withRouter(Container);
