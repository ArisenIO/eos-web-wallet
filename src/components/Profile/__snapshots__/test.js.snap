// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders without crashing 1`] = `
<div
  className="box"
>
  <div
    className="level"
  >
    <div
      className="level-left"
    >
      <div
        className="level-item"
      >
        <img
          alt="Rick Sanchez"
          className={undefined}
          src="src"
        />
      </div>
      <div
        className="level-item"
      >
        <div>
          <div
            className="heading is-6"
          >
            Rick Sanchez
          </div>
          <div
            className="title"
          >
            Friday Harbor, WA
          </div>
          <a
            href="http://twitter.com/"
          >
            http://twitter.com/
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="u-mt2"
  >
    <p>
      active
    </p>
  </div>
</div>
`;
